Sources/LeafErrorMiddleware/LeafErrorMiddleware.swift

```diff
@@ -1,61 +1,79 @@
+    public init(environment: Environment, log: Logger) {
         self.environment = environment
-        self.viewRenderer = viewRenderer
+        self.log = log
     }
 
-    public func respond(to req: Request, chainingTo next: Responder) throws -> Response {
-        do {
-            return try next.respond(to: req)
-        } catch {
-            log.error(error)
-            return make(with: req, for: error)
-        }
-    }
+    /// See `Middleware.respond`
+    public func respond(to req: Request, chainingTo next: Responder) throws -> Future<Response> {
+        let renderer = try req.make(LeafRenderer.self)
+        let promise = req.eventLoop.newPromise(Response.self)
+
+        func handleError(with status: HTTPStatus) {
+            if status == .notFound {
+                do {
+                    try renderer
+                        .render("404")
+                        .encode(for: req)
+                        .do { res in
+                            res.http.status = status
+                            promise.succeed(result: res)
+                    }
+                } catch { /* swallow so we return the default view */ }
+            }
 
-    public func make(with req: Request, for error: Error) -> Response {
-        let status: Status = Status(error)
-        if status == .notFound {
             do {
-                let response = try viewRenderer.make("404", Node([:])).makeResponse()
-                response.status = .notFound
-                return response
-            } catch { /* swallow so we return the default view */ }
+                let parameters: [String:String] = [
+                    "status": status.code.description,
+                    "statusMessage": status.reasonPhrase
+                ]
+                try renderer
+                    .render("serverError", parameters)
+                    .encode(for: req)
+                    .do { res in
+                        res.http.status = status
+                        promise.succeed(result: res)
+                }
+            } catch {
+                let body = "<h1>Internal Error</h1><p>There was an internal error. Please try again later.</p>"
+                try! body.encode(for: req).do { res in
+                    res.http.status = status
+                    res.http.headers.replaceOrAdd(name: .contentType, value: "text/html; charset=utf-8")
+                    promise.succeed(result: res)
+                }
+            }
         }
-        
+
         do {
-            let parameters: [String: NodeRepresentable] = [
-                "status": status.statusCode.description,
-                "statusMessage": status.reasonPhrase
-            ]
-            let response = try viewRenderer.make("serverError", parameters).makeResponse()
-            response.status = status
-            return response
+            try next.respond(to: req).do { res in
+                if res.http.status != .ok {
```

Comment:
Not found errors won't throw, this is the only way to catch them

Reply:
`Abort.notFound` errors should definitely throw - if that's not the case then there is a problem

Reply:
If I'm reading this correctly then any other status that isn't a 200 would cause the error page to be rendered? So any redirects, a 201, etc?

Reply:
Yeah, I added it only to check. But while testing, no errors where thrown, only http.response with 404 status 😕

Reply:
See example: https://github.com/bre7/leaferror-test. Code inside `catch` is never called

Reply:
Hmm it may be a change in behaviour - I'm going to have to dig into this to see what should be expected